rust/src/passwd.rs

```diff
@@ -389,6 +389,40 @@ fn write_data_from_treefile(
         })
         .with_context(|| format!("failed to write /{}", &target_etc_filename))?;
 
+    // Regernate etc/{,g}shadow to sync with etc/{password,group}
+    let db = rootfs.open(target_etc_filename).map(BufReader::new)?;
+    let shadow_name = if target == "passwd" {
+        "shadow"
+    } else {
+        "gshadow"
+    };
+    let target_etc_shadow = format!("{}{}", dest_path, shadow_name);
+
+    match shadow_name {
+        "shadow" => {
+            let entries = nameservice::passwd::parse_passwd_content(db)?;
+            rootfs
+                .atomic_replace_with(&target_etc_shadow, |target_shadow| -> Result<()> {
+                    for user in entries {
+                        target_shadow.write_fmt(format_args!("{}:*::0:99999:7:::\n", user.name))?;
```

Comment:
This is more idiomatically `writeln!(target_shadow, "{}:*::0:99999:7:::", user.name)?;`  (Notice I dropped the literal `\n` in favor of `ln!` in `writeln!`)